lineax/_solver/gmres.py

```diff
@@ -395,13 +400,17 @@ def _arnoldi_gram_schmidt(
 
     def _normalise(
         self, x: PyTree[Array], eps: Optional[Float[ArrayLike, ""]]
-    ) -> tuple[PyTree[Array], Float[Array, ""], Bool[ArrayLike, ""]]:
+    ) -> tuple[PyTree[Array], Inexact[Array, ""], Bool[ArrayLike, ""]]:
+        input_dtype = jnp.result_type(*jtu.tree_leaves(x))
         norm = two_norm(x)
         if eps is None:
             eps = jnp.finfo(norm.dtype).eps
+        else:
+            eps = jnp.astype(eps, jnp.finfo(input_dtype).dtype)
```

Comment:
I think this should probably just use `norm.dtype`?

Reply:
Yes